### PR TITLE
Rename and move updateItem #8653

### PR DESCRIPTION
**Pull request by mshabarov**
Part of #8653

### CODE REVIEW

File: flow-data/src/main/java/com/vaadin/flow/data/provider/AbstractDataView.java [hunk 1]
```diff
@@ -101,6 +104,13 @@ protected final void verifyDataProviderType(Class<?> dataProviderType) {
         return dataProviderSupplier.get().fetch(new Query<>());
     }
 
+    @Override
+    public void refreshItem(T item) {
+        Objects.requireNonNull(item, NULL_ITEM_ERROR_MESSAGE);
+        getItems().filter(i -> equals(item, i)).findFirst()
+                .ifPresent(i -> dataProviderSupplier.get().refreshItem(i));
```
Comment (pleku) [inline flow-data/src/main/java/com/vaadin/flow/data/provider/AbstractDataView.java:111]: Why is this getting the old item and then calling refresh with that ???
If I look at the data provider & communicator, the latter will use the item provided by the former in the event and then update using that one - so basically this code here, to my understanding, will loose all the changes when the objects are different but even if they have matching identifiers.

So to explain this:
old bean: Person:{ id=0, name="Liam" }
new bean Person:{ id=0, name="William" } 
identifier is id field.
When given the new bean to refresh, this code finds the old bean using the identifier.
Then it passes the old bean to data provider which fires the event.
The data communicator gets the event and uses the old bean for refreshing things and the update is missed.

Or am I missing something here ?

File: flow-data/src/main/java/com/vaadin/flow/data/provider/DataView.java [hunk 2]
```diff
@@ -42,6 +42,26 @@
...
+     * <p>
+     * This method delegates the update to
+     * {@link DataProvider#refreshItem(Object)}.
+     *
+     * @param item
+     *            item containing updated state
+     *
+     * @see #setIdentifierProvider(IdentifierProvider)
+     */
+    void refreshItem(T item);
```
Comment (pleku) [inline flow-data/src/main/java/com/vaadin/flow/data/provider/DataView.java:63]: I would like to have a unit test using the `AbstractLazyDataView` where it is tested by minimum that calling the `refreshItem` will trigger something in the `DataCommunicator` - mocking or spying can be used to verify this.

File: flow-data/src/main/java/com/vaadin/flow/data/provider/AbstractDataView.java (hunk 1 above)
Comment (mshabarov) [reply flow-data/src/main/java/com/vaadin/flow/data/provider/AbstractDataView.java:111]: Well, seems like I overlooked the use cases on `data communicator` side. At least in `getJsonItems` and well-known `getItem` uses the new object passed to `refreshItem` method. So, yes, should be fixed.

Comment (mshabarov) [reply flow-data/src/main/java/com/vaadin/flow/data/provider/AbstractDataView.java:111]: Oh, wait... is this a proper use of `refreshItem`?  
```
refreshItem(new Person(0));
```
At least `CheckboxGroup` uses it in it's tests. It seems that there should be a two rules for `refreshItem`:

1. Developer should change the exact object which he obtained previously from data set somehow.
2. Developer should pass this ref to changed object to `refreshItem`.

This means `refreshItem(new Person(0))` is prohibited.

File: flow-data/src/main/java/com/vaadin/flow/data/provider/DataView.java (hunk 2 above)
Comment (mshabarov) [reply flow-data/src/main/java/com/vaadin/flow/data/provider/DataView.java:63]: Done

**Pull request merged**